### PR TITLE
add snowline Laplace approximation and Importance Sampling python package for Bayesian inference

### PR DESCRIPTION
**Pull request by JohannesBuchner**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by JohannesBuchner** [comment]
I am seeing this error:

```
Reloading output folder: /home/conda/staged-recipes/build_artifacts
Mamba failed to solve:
 - pypmc
 - python 3.9.* *_cpython
 - pytest-runner
 - numpy 1.20.*
 - pip
 - iminuit

with channels:
 - local
 - conda-forge
 - conda-forge

The reported errors are:
- Encountered problems while solving:
-   - package pypmc-1.2-py37hc105733_0 requires python_abi 3.7.* *_cp37m, but none of the providers can be installed
```

pypmc is a cpython package that snowline depends on.

I don't understand why "python 3.9.* *_cpython" tries to pull in the py37h version of pypmc.

iminuit is another cpython package that snowline depends on. Its recipe is here: https://github.com/conda-forge/iminuit-feedstock/blob/main/recipe/meta.yaml

This [stackoverflow question](https://stackoverflow.com/questions/73433762/why-does-conda-package-install-complain-about-both-python-abi-3-7-and-3-9) has the same error but from a user perspective.

snowline is pure python, but requires the two cpython packages.

@conda-forge/help-python-c, can you please help?

File: recipes/snowline/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    - pypmc
+    - iminuit
+
+test:
+  imports:
+    - snowline
+  source_files:
+    - tests/*
+  requires:
+    - pytest
```
Comment (carterbox) [inline recipes/snowline/meta.yaml:41]:
```suggestion
test:
  imports:
    - snowline
  source_files:
    - tests/*
  requires:
    - pytest >=3
    - pip
  commands:
    - pip check
    - pytest tests/
```

File: recipes/snowline/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    - numpy
+    - pytest-runner
+
+  run:
+    - python
+    - numpy
+    - matplotlib-base
+    - corner
+    - pypmc
+    - iminuit
```
Comment (carterbox) [inline recipes/snowline/meta.yaml:33]: These do not match the upstream metadata

https://github.com/JohannesBuchner/snowline/blob/6f9b5f767903ebf0537569ff85752a8533664b8a/setup.py#L23

File: recipes/snowline/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - python
+    - pip
+    - pypmc
+    - iminuit
+    - numpy
+    - pytest-runner
```
Comment (carterbox) [inline recipes/snowline/meta.yaml:25]:
```suggestion
  host:
    - python
    - pytest-runner
    - setuptools
```

**Review by carterbox — CHANGES_REQUESTED** [review]
If this package is pure python, then it can be `noarch: python`.

File: recipes/snowline/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - python
+    - pytest-runner
+    - setuptools
+
+  run:
+    - python
```
Comment (carterbox) [inline recipes/snowline/meta.yaml:25]:
```suggestion
build:
  number: 0
  noarch: python
  script: {{ PYTHON }} -m pip install . -vv

requirements:
  host:
    - python >=3.5
    - pytest-runner
    - setuptools
  run:
    - python >=3.5
```

**Review by carterbox — CHANGES_REQUESTED** [review]
Figure out the tests and fix the incorrect runtime requirements.

**Comment by carterbox** [comment]
> Tests are fixed by installing openssh to make openmpi work.

Looks like openssh is only required because these packages are being built inside a container? It's not actually a direct dependency of your package. Only direct dependencies of your package go in the requirements section. Packages that are only required for the tests go in the test/requires section, not the requirements/run section.

**Comment by JohannesBuchner** [comment]
> > Tests are fixed by installing openssh to make openmpi work.
> 
> Looks like openssh is only required because these packages are being built inside a container? It's not actually a direct dependency of your package. Only direct dependencies of your package go in the requirements section. Packages that are only required for the tests go in the test/requires section, not the requirements/run section.

Thank you for pointing this out! I made the change.

**Comment by JohannesBuchner** [comment]
I think this is ready to be merged

**Pull request merged**